### PR TITLE
Implement follow & unfollow mutations

### PR DESCRIPTION
**Pull request by dishanest**
## Overview
Implemented follow and unfollow mutations

## Changes Made
- Added GraphQL mutations
  - Contain `followedPublications` field for debugging
  - Response is a `User` type (due to backend schema)
- Added error message for null UUID
- Update cache only after follow/unfollow operation successfully return

## Test Coverage
- Verified `followedPublicationIDs` (both cached and from network response) changes accordingly using prints before & after mutation

### CODE REVIEW

File: Volume/Supporting/UserData.swift [hunk 1]
```diff
@@ -120,12 +120,31 @@ class UserData: ObservableObject {
...
+        guard let uuid = UserDefaults.standard.string(forKey: userUUIDKey) else {
+            print("Error: failed to set follow status because user UUID not set")
+            return
+        }
+        
         if isFollowed {
-            if !followedPublicationIDs.contains(publication.id) {
-                followedPublicationIDs.insert(publication.id, at: 0)
-            }
+            cancellables[.follow(publication)] = Network.shared.publisher(for: FollowPublicationMutation(publicationID: publication.id, uuid: uuid))
```
Comment (danielVebman) [inline Volume/Supporting/UserData.swift:129]: long line

File: Volume/Supporting/UserData.swift [hunk 2]
```diff
@@ -120,12 +120,31 @@ class UserData: ObservableObject {
...
+                .sink { completion in
+                    if case let .failure(error) = completion {
+                        print(error)
+                    }
+                } receiveValue: { value in
+                    if !self.followedPublicationIDs.contains(publication.id) {
+                        self.followedPublicationIDs.insert(publication.id, at: 0)
+                    }
```
Comment (danielVebman) [inline Volume/Supporting/UserData.swift:137]: I've got my 4410 hat on: This isn't thread safe. What if you spam the follow button. Is it possible that you end up seeing the same publication id twice? Probably not in reality, but still.

File: Volume/Supporting/UserData.swift [hunk 3]
```diff
@@ -120,12 +120,31 @@ class UserData: ObservableObject {
...
+                }
         } else {
-            followedPublicationIDs.removeAll(where: { $0 == publication.id })
+            cancellables[.unfollow(publication)] = Network.shared.publisher(for: UnfollowPublicationMutation(publicationID: publication.id, uuid: uuid))
+                .sink { completion in
+                    if case let .failure(error) = completion {
+                        print(error)
+                    }
+                } receiveValue: { _ in
+                    self.followedPublicationIDs.removeAll(where: { $0 == publication.id })
```
Comment (danielVebman) [inline Volume/Supporting/UserData.swift:146]: Similar to my comment above: Swift arrays are not thread safe.

**Review by danielVebman — APPROVED** [review]
2/3 comments! Good work pushing out these PRs

**Review by amyqxhuang — APPROVED** [review]
looks good!

**Review by NgaiJustin — APPROVED** [review]
looks good to me.

**Review by spd75 — APPROVED** [review]
Lgtm

**Pull request merged**